File: backend/app/schemas/report_search.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel


SortOption = Literal["newest", "oldest", "score_desc", "score_asc", "name_asc", "name_desc"]
QuickRange = Literal["today", "week", "month"]


class ReportSearchItem(BaseModel):
    id: int
    file_name: str
    patient_name: str
    report_type: str | None = None
    health_score: int
    overall_status: str
    risk_level: str
    summary_snippet: str | None = None
    created_at: datetime


class ReportFilterOptions(BaseModel):
    # Populated from the user's own real data — never a hardcoded list.
    risk_levels: list[str]
    overall_statuses: list[str]
    report_types: list[str]


class ReportSearchResponse(BaseModel):
    items: list[ReportSearchItem]
    total: int
    page: int
    page_size: int
    total_pages: int
    filter_options: ReportFilterOptions

File: frontend/src/types/reportSearch.ts
export type SortOption = "newest" | "oldest" | "score_desc" | "score_asc" | "name_asc" | "name_desc";
export type QuickRange = "today" | "week" | "month";

export interface ReportSearchItem {
  id: number;
  file_name: string;
  patient_name: string;
  report_type: string | null;
  health_score: number;
  overall_status: string;
  risk_level: string;
  summary_snippet: string | null;
  created_at: string;
}

export interface ReportFilterOptions {
  risk_levels: string[];
  overall_statuses: string[];
  report_types: string[];
}

export interface ReportSearchResponse {
  items: ReportSearchItem[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
  filter_options: ReportFilterOptions;
}

export interface ReportSearchParams {
  q?: string;
  risk_level?: string[];
  overall_status?: string[];
  report_type?: string[];
  min_score?: number;
  max_score?: number;
  date_from?: string;
  date_to?: string;
  quick_range?: QuickRange;
  sort?: SortOption;
  page?: number;
  page_size?: 10 | 20 | 50 | 100;
}

export const DEFAULT_SEARCH_PARAMS: ReportSearchParams = {
  sort: "newest",
  page: 1,
  page_size: 20,
};